src/driver.rs

```diff
@@ -151,4 +152,22 @@ impl ClintDriver {
             self.write_msip(i, 1).unwrap();
         }
     }
+
+    /// Create a pending MSI interrupts for each harts of the platform, except the current one.
+    #[allow(dead_code)]
+    pub fn trigger_msi_on_all_other_harts(&mut self) {
+        let current_hart: usize;
+        unsafe {
+            asm!(
+            "csrr {0}, mhartid",
+            out(reg) current_hart
+            );
+        }
```

Comment:
We don't use assembly outside of the `arch` module. This is important as this basically prevent from compiling the module to another architecture, which we need for unit tests and formal verification with the Sail model in the future.

Instead you can use the `read_csr` function from `Arch` to get the current hart ID.

Reply:
I believed it was only for virt.rs. Ma bad, sorry about that! Fixed now!